### DISCUSSION TITLE
dbRDA Plot Only Showing First 3 Arrows Instead of All Variables?

### DISCUSSION THREAD

**Discussion by timz0605** [vegandevs/vegan#671] [Q&A]
Hello everyone!

I've been trying to plot my dbRDA (or CCA) results after the analyses, but am having trouble showing all the arrows for environmental variables. In total, I have approx. 10 environmental variables that I want to see how that might explain the changes in the community compositions. I have tried `plot`, `autoplot`, and also some other methods, yet all the plotting methods would only show the first three arrows instead of all. I was wondering if anyone have encountered similar issues before?

Below is the code I use to generate the plot:

```
env_data_mean <- read.csv('env_data_mean.csv', 
                     header = TRUE, 
                     check.names = FALSE, 
                     row.names = 1)

freqTable_106 <- read.table('zotutable_output_a5_106.txt', header = TRUE, sep = '\t', 
                        row.names = 1, check.names = FALSE, comment.char = '')

env_var <- env_data_mean %>% 
  select(c(BOD, Chlo, DO, FC, PhP, Sal, TN, Turb, VSS))

ARMS_dbrda <- dbrda(freqTable_106 ~ BOD + Chlo + DO + FC + PhP + Sal + TN + Turb + VSS, 
                    data = env_var, dist = "bray")

autoplot(ARMS_dbrda)

plot(ARMS_dbrda)
```
And this is the plot I am getting. No matter what I did, only the first three variables, `BOD`, `Chlo`, and `DO` would show up. Interestingly, if I delete those three, then the next three, which are `FC`, `PhP`, and `Sal` would then show up.
![image](https://github.com/user-attachments/assets/da385383-825f-4751-81f4-c90a7d66cdec)

Thank you!

**Comment by gavinsimpson** [comment]
Without your data or the output from `print(ARMS_dbrda)`, I would guess that your constraints (environmental variables) are *strongly* collinear to the extent that beyond the first three, the remaining variables provide no unique information to the constraint data and are thus aliased. If so, this should be noted in the output when you print the fitted dbRDA.

**Comment by jarioksa** [comment]
This indeed seems to be the same user issue that has been answered in #658, #582, #511, #506 and #456 to list some of the more recent ones (there are many more, and still more in StackOverflow).

**Reply by gavinsimpson** [reply]
I wonder if we need to make this more clear to users? What about adding a `message()` (`warning()? Might be too aggressive?) call that outputs during fitting to warn the user about the aliased (missing) variables?

**Reply by jarioksa** [reply]
Yes, I think so. When I initially wrote the functions, I thought that warning will disappear after being issued, and it is better to have a permanent reminder. The users have changed, and it seems that now people just copy-paste "scripts" and never stop to inspect the result object before advancing to the next step. I already added a `message()` about overfitted model (no residual component), and having a `message()` of aliasing surely is useful (in addition to the permanent message in `print`).

**Reply by gavinsimpson** [reply]
Yeah; I think a message in addition to the permanent message in the printed output is the right solution at this stage.

**Reply by jarioksa** [reply]
`ordConstrained()` is probably the best place, because `rda`, `cca`, `capscale` and `dbrda` will all go through this.

It seems that I used there `warning()` instead of `message()`  of overfitting, but was not quite sure which one would be better as seen in commit message of f8f5221487cb247a75a4d6ac22f7bfd29c0d87a6. Both things should be made the same, and `message()` may indeed be better.

**Reply by jarioksa** [reply]
@gavinsimpson do you want to do this, or should I do this change? Your English is better than mine so you could at least suggest text. Now I think we should have the following:
- `ordConstrain` should issue a `message()` of aliased variables (and preferably in plain English: the current one of aliasing sounds complicated).
- `ordConstrain` should have a `message()` instead of current `warning()` of overfitted models and missing residual component.
- `print.cca` should print a text of missing residual component (and overfitted) model in addition to the current text of aliased redundant constraints and/or conditions.

**Reply by gavinsimpson** [reply]
Yep, I'll do this one @jarioksa and then do a PR for comment.
